features/enlinkd/tests/src/test/java/org/opennms/netmgt/enlinkd/Nms17216EnIT.java

```diff
@@ -745,10 +748,142 @@ public void testNetwork17216CdpTopology() throws Exception {
 
         assertTrue(m_linkd.runSingleSnmpCollection(switch1.getId()));
         assertEquals(5, m_cdpLinkDao.countAll());
+        assertTrue(m_cdpTopologyService.hasUpdates());
         
         assertTrue(m_linkd.runSingleSnmpCollection(switch2.getId()));
         assertEquals(11, m_cdpLinkDao.countAll());
+        assertTrue(m_cdpTopologyService.hasUpdates());
+        
+        // Now we support by default configuration: BRIDGE, CDP, ISIS, LLDP and OSPF
+        assertEquals(5, m_topologyDao.getSupportedProtocols().size());
+        assertTrue(m_topologyDao.getSupportedProtocols().contains(ProtocolSupported.BRIDGE.name()));
+        assertTrue(m_topologyDao.getSupportedProtocols().contains(ProtocolSupported.CDP.name()));
+        assertTrue(m_topologyDao.getSupportedProtocols().contains(ProtocolSupported.ISIS.name()));
+        assertTrue(m_topologyDao.getSupportedProtocols().contains(ProtocolSupported.LLDP.name()));
+        assertTrue(m_topologyDao.getSupportedProtocols().contains(ProtocolSupported.OSPF.name()));
+
+        m_cdpTopologyService.updatesAvailable();
+        assertEquals(2, m_nodeTopologyService.findAllSnmpNode().size());
+        assertEquals(2, m_nodeTopologyService.findAllNode().size());
+        assertEquals(6, m_nodeTopologyService.findAllIp().size());
+        
+        m_nodeTopologyService.findAllNode().stream().forEach( node -> {
+            System.err.println(node);
+            assertNotNull(node.getId());
+            assertNotNull(node.getLabel());
+        });
+
+        m_nodeTopologyService.findAllIp().stream().forEach( ip -> {
+            System.err.println(ip);
+            assertNotNull(ip.getId());
+            assertNotNull(ip.getIpAddress());
+            assertNotNull(ip.getIsSnmpPrimary());
+            assertTrue(ip.isManaged());
+        });
+
+        CdpOnmsTopologyUpdater cdptopology = m_linkd.getCdpTopologyUpdater();
+        assertNotNull(cdptopology);
+        OnmsTopology topology = cdptopology.buildTopology();
+        assertEquals(2, topology.getVertices().size());
+        assertEquals(4, topology.getEdges().size());
+        
+        OnmsTopologyLogger tl = EnLinkdBuilderITCase.createAndSubscribe(
```

Comment:
Can we write a test that asserts that the `OnmsTopologyConsumer` has actually received (or consumed) the expected messages?

Reply:
commit 251ee68c19266e247bd9d84ea709a67ca2ebae11 (HEAD -> jira/HZN-1393)
Author: Antonio <rssntn67@yahoo.it>
Date:   Wed Jan 16 21:22:48 2019 +0100

    OnmsTopologyLogger has actually received the expected messages